base/broadcast.jl

```diff
@@ -280,7 +280,10 @@ end
 eltypestuple(a) = (Base.@_pure_meta; Tuple{eltype(a)})
 eltypestuple(T::Type) = (Base.@_pure_meta; Tuple{Type{T}})
 eltypestuple(a, b...) = (Base.@_pure_meta; Tuple{eltypestuple(a).types..., eltypestuple(b...).types...})
-_broadcast_eltype(f, A, Bs...) = Base._return_type(f, eltypestuple(A, Bs...))
```

Comment:
I'd leave this as is, and make the change [here](https://github.com/JuliaLang/julia/blob/4331b2989701324334b51d237f7dd833b5f439ae/base/broadcast.jl#L294) to `return similar(Array{T === Union{} ? Any : T}, shape)`